### PR TITLE
added selection support

### PR DESCRIPTION
**Pull request by basisunus**
Only widgets are selected by holding down the shift while clicking.

### CODE REVIEW

File: bin/.gitignore [hunk 1]
```diff
@@ -1 +0,0 @@
-*
```
Comment (dcwhite) [inline bin/.gitignore]: Don't delete this file! Please revert.

**Comment by dcwhite** [comment]
I'll look and then @aylakhan should look.

File: bin/.gitignore (hunk 1 above)
Comment (basisunus) [inline bin/.gitignore]: not sure when and how I deleted it. How can add it back?

Comment (dcwhite) [inline bin/.gitignore]: You must have deleted all of the bin/ directory. You can checkout this version, then recreate bin/ and add the .gitignore back, then fast forward your HEAD and commit it back.
https://github.com/basisunus/SCIRun/blob/c24fa28687b21f646c8155fa0d8b104473d1239b/bin/.gitignore

Comment (dcwhite) [inline bin/.gitignore]: Or just recreate manually, it's simply a one-line text file with "*" in it

**Comment by dcwhite** [comment]
Thanks. I ran this and was able to generate console messages when shift-clicking some widgets. They don't change color yet. Also, I started getting a lot of `Unable to find uniform with name:...` messages too, when I shift-clicked outside of the widget in ViewScene.

**Comment by dcwhite** [comment]
@aylakhan can you test as well? I'll add some code comments later today.

**Comment by aylakhan** [comment]
Mac OS X: tried SHIFT-clicking on probe from GenerateSinglePointProbeFromField in ViewScene, but I'm not seeing any console messages.

**Comment by basisunus** [comment]
I need a mac computer to see what the problem is

**Comment by aylakhan** [comment]
There's a Mac computer in the CIBC lab on the 4th floor that you're welcome to use.

**Comment by basisunus** [comment]
There are two Macs. Which one should I use?

**Comment by dcwhite** [comment]
The "bigger" one is more up to date--10.7 I believe..called "Gus"

**Comment by dcwhite** [comment]
I confirm @aylakhan's test on my Mac (10.8, NVidia). I get the "shader missing" messages when shift-clicking but no selection messages.

**Comment by basisunus** [comment]
how can i see the terminal output?

**Comment by dcwhite** [comment]
If you run `SCIRun_test` from the terminal, it just gets printed to the terminal.

**Comment by basisunus** [comment]
It's working on both Windows and OSX now. Shift-click to select and change color to red.

**Comment by aylakhan** [comment]
OK, I'm seeing the color change. I'm also getting some output (from the shader?):

Unable to find uniform with name: uAmbientColor in shader with ID: 10
Unable to find uniform with name: uDiffuseColor in shader with ID: 10
Unable to find uniform with name: uSpecularColor in shader with ID: 10
Unable to find uniform with name: uSpecularPower in shader with ID: 10
Unable to find uniform with name: uAmbientColor in shader with ID: 10
Unable to find uniform with name: uDiffuseColor in shader with ID: 10
Unable to find uniform with name: uSpecularColor in shader with ID: 10
Unable to find uniform with name: uSpecularPower in shader with ID: 10
Unable to find uniform with name: uAmbientColor in shader with ID: 10
Unable to find uniform with name: uDiffuseColor in shader with ID: 10
Unable to find uniform with name: uSpecularColor in shader with ID: 10
Unable to find uniform with name: uSpecularPower in shader with ID: 10
Unable to find uniform with name: uAmbientColor in shader with ID: 10
Unable to find uniform with name: uDiffuseColor in shader with ID: 10
Unable to find uniform with name: uSpecularColor in shader with ID: 10
Unable to find uniform with name: uSpecularPower in shader with ID: 10
Unable to find uniform with name: uAmbientColor in shader with ID: 10
Unable to find uniform with name: uDiffuseColor in shader with ID: 10
Unable to find uniform with name: uSpecularColor in shader with ID: 10
Unable to find uniform with name: uSpecularPower in shader with ID: 10
Shader GC: Shaders/Selection

**Comment by basisunus** [comment]
I didn't find a way to remove those.

**Comment by aylakhan** [comment]
Are there any files that should go in the Assets or Shaders directories?

**Comment by basisunus** [comment]
Selection.vs and Selection.fs for shaders. No assets.

**Comment by aylakhan** [comment]
Do you see this output (or similar) on Windows?

**Comment by dcwhite** [comment]
Yes, that's what we were talking about in the meeting. It will be a performance drag with lots of widgets.

**Comment by basisunus** [comment]
Yes. They are pretty much the same.

**Comment by aylakhan** [comment]
I'll merge, but let's see if we can track down the source of those messages. Also, Linux will need to be tested.

**Comment by basisunus** [comment]
I just feel the rendering system doesn't provide a good mechanism for selection/feedback. A perfect solution may need some work on the basic system.

**Pull request merged**